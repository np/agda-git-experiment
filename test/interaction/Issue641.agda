module Issue641 where
Foo : Set
Foo = Set
