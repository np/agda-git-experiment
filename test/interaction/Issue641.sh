# This test tries to ensure that (certain) changes to OPTIONS pragmas
# are respected when a file is reloaded.

BASE=Issue641
FAIL="module $BASE where\nFoo : Set\nFoo = Set\n"
SUCCEED="{-# OPTIONS --type-in-type #-}\n$FAIL"
LOAD="IOTCM \"$BASE.agda\" None (Cmd_load \"$BASE.agda\" [])\n"

rm -f $BASE.pipe
mkfifo $BASE.pipe

agda --interaction < $BASE.pipe 2>&1 &

( (printf "$SUCCEED" > $BASE.agda); printf "$LOAD"; \
  sleep 1; \
  (printf "$FAIL" > $BASE.agda); printf "$LOAD" \
) > $BASE.pipe

rm -f $BASE.pipe
