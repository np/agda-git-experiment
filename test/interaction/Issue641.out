Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue641 (Issue641.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished Issue641.\n" t)
(agda2-status-action "Checked")
(agda2-info-action "*All Goals*" "" nil)
((last . 1) . (agda2-goals-action '()))
Agda2> (agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking Issue641 (Issue641.agda).\n" t)
(agda2-info-action "*Error*" "Issue641.agda:3,7-10\nSet₁ != Set\nwhen checking that the expression Set has type Set" nil)
((last . 3) . (agda2-goto '("Issue641.agda" . 39)))
(agda2-status-action "")
Agda2> 